### PR TITLE
enforce cookie consent policy

### PR DESCRIPTION
**Pull request by roymiloh**

### CODE REVIEW

File: src/data-capsule.js [hunk 1]
```diff
@@ -22,7 +25,7 @@ export default class DataCapsule extends BaseStorage {
     return options;
   }
 
-  setItem(key, value, options) {
+  async setItem(key, value, options) {
```
Comment (splintor) [inline src/data-capsule.js:28]: Why is this needed? 🤔

File: src/strategies/local-storage.js [hunk 2]
```diff
@@ -49,16 +50,21 @@ function updateAccessTime(fullKey, data) {
 }
 
 export default class LocalStorageStrategy extends BaseStorage {
-  setItem(key, value, options) {
+  async setItem(key, value, options) {
```
Comment (splintor) [inline src/strategies/local-storage.js:53]: Why is this needed? 🤔

File: test/strategies/in-memory-storage.spec.js [hunk 3]
```diff
@@ -10,9 +10,9 @@ describe('in-memory-strategy', () => {
     );
   });
 
-  it('should throw if no namespace in setItem', () => {
+  it('should throw if no namespace in setItem', async () => {
     const capsule = new InMemoryStorageCapsule();
-    expect(() => capsule.setItem('shahata', 123)).to.throw(
+    await expect(capsule.setItem('shahata', 123)).to.eventually.rejectedWith(
```
Comment (splintor) [inline test/strategies/in-memory-storage.spec.js:15]: Why is this change needed? `setItem` is now defined as `async`, but actually, it contains no `await` and doesn't return a promise, so it is not really `async`, and this code doesn't need to change.

File: src/data-capsule.js (hunk 1 above)
Comment (roymiloh) [reply src/data-capsule.js:28]: Because `setItem` is an async operation, that's the contract. Before that change it was very bad, it could `throw` out of the promise chain, and **just then** it returned a promise. Note that as a part of the commit I removed the `return Promise.resolve()` line (from local storage strategy) and added an `async` annotation instead.

File: src/data-capsule.js [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
+import { verifyConsentPolicyCategoryIfExists } from './consent-policy';
 
 function validateNamespace(options) {
   if (!options.namespace) {
     throw new Error('namespace is required');
   } else if (typeof options.namespace !== 'string') {
     throw new Error('namespace should be a string');
   }
+
+  verifyConsentPolicyCategoryIfExists(options.category);
```
Comment (splintor) [inline src/data-capsule.js:11]: If `category` is (about to be) an expected field in `options`, it should be added to `Options` in `src/typings/index.d.ts`.

File: src/data-capsule.js (hunk 1 above)
Comment (splintor) [reply src/data-capsule.js:28]: Yes, I didn't fully realize until now that `async` annotation causes a function to return a promise even if it didn't include an explicit `return` statement.

Thanks!

File: test/strategies/in-memory-storage.spec.js (hunk 3 above)
Comment (splintor) [reply test/strategies/in-memory-storage.spec.js:15]: Nevermind, I now realized that `async` changes what `setItem` returns, and require changing the test.

File: src/data-capsule.js (hunk 4 above)
Comment (roymiloh) [reply src/data-capsule.js:11]: Correct, I guess I just forgot:) but let’s call it a feature - afaik it shouldn’t be exposed _yet_.
